### DISCUSSION TITLE
NoiseTorch unconfigured?

### DISCUSSION THREAD

**Discussion by Doug140** [noisetorch/NoiseTorch#326] [General]
I have a message at the top of the NoiseTorch window that says "NoiseTorch unconfigured." Everything seems to be working normally. Is there something I still need to configure, or is this a bug?

![Screenshot_20220607_061425](https://user-images.githubusercontent.com/63794602/172479954-a3cc40cd-d885-4139-8c95-1788a37a7eba.png)

**Comment by AXDOOMER** [comment]
It's an issue since https://github.com/noisetorch/NoiseTorch/commit/102d6fd57ac6ec6d847fb1ddf24d54522ce12111 and/or https://github.com/noisetorch/NoiseTorch/commit/f2d726a9cff2e7cc97c55cecd4748aaad905f12c. 

Even though you may be using your device (in Discord or elsewhere), if it's not selected in Pulse Audio, it shows unconfigured. 

![Peek 2022-06-07 17-08](https://user-images.githubusercontent.com/6194072/172483152-52e325d3-7f21-460a-b6ac-098a9e5e6cbd.gif)

**Reply by TheDukeofErl** [reply]
Interesting. I don't have this issue with pipewire and don't have any systems running pulseaudio so I'm not surprised I missed it during testing. I'd best that it's this if specifically: https://github.com/noisetorch/NoiseTorch/blob/master/ui.go#L85. 

Pulseaudio must be handling things differently under the hood compares to pipewire and a different method of detecting if the state is unconfigured may be necessary on both. I suppose that's not super surprising but it is a bit disappointing in this case.

I'll have to spin up a VM or something to take a look. Hopefully Ubuntu 22.04 (or something else that doesn't require much configuration on my end) still uses pulseaudio out of the box.

**Reply by AXDOOMER** [reply]
I run Ubuntu 22.04 LTS

**Reply by RiemaruKarurosu** [reply]
In case this help, replicated issue, it seems it's only with pulseaudio
![Screenshot_20220612_093507](https://user-images.githubusercontent.com/45057400/173238363-8b2a28c3-6f06-4778-8a55-317ef592b52c.png)
m

**Reply by TheDukeofErl** [reply]
Thanks for confirming. Hopefully I'll have some time on my end to spin up a VM to work out a solution soon.

**Comment by Doug140** [comment]
It shows unconfigured even when I have the NoiseTorch mic. selected in PulesAudio.

**Comment by Doug140** [comment]
In case it helps, here's my system information. Feren OS is based on Ubuntu.

![Screenshot_20220607_173433](https://user-images.githubusercontent.com/63794602/172494428-ad8de51a-5f63-430a-86c9-8b13878064bf.png)

**Comment by TheDukeofErl** [comment]
Created https://github.com/noisetorch/NoiseTorch/issues/340 to track this issue. Going to go ahead and lock this, if anyone would like to add anything else feel free to do so over there!
